src/catalogue.js

```diff
@@ -124,13 +124,19 @@ export function loadDateDataSets(projectId, options) {
         bucketItems = bucketItemsToExecConfig(bucketItems, { removeDateItems: true });
     }
 
+    const omittedOptions = ['filter', 'types', 'paging', 'dataSetIdentifier', 'returnAllDateDataSets', 'returnAllRelatedDateDataSets'];
+    // includeObjectsWithTags has higher priority than exludeObjectsWithTags, so when present omit exludeObjectsWithTags
```

Comment:
maybe we could add a test here https://github.com/gooddata/gooddata-js/blob/master/test/catalogue_test.js to have the desired behaviour described by a test.